ssg/utils.py

```diff
@@ -333,7 +333,16 @@ def enum(*args):
     return type('Enum', (), enums)
 
 
-def apply_formatting_on_dict_values(source_dict, string_dict, ignored_keys=[]):
+def recurse_or_substitute_or_do_nothing(v, string_dict, ignored_keys):
```

Comment:
```suggestion
def recurse_or_substitute_or_do_nothing(v, string_dict, ignored_keys=frozenset()):
```
This way you won't have to check for None. It'll retain the iterable type and won't be affected by the "mutable default argument" problem.